### DISCUSSION TITLE
Repeated center half should behave same as repeated left/right half

### DISCUSSION THREAD

**Discussion by mastef** [rxhanson/Rectangle#383] [Ideas]
When I press "left half" or "right half" multiple times, it toggles through 3 different widths. When I press "center half" multiple times, nothing happens - the width stays unchanged.

It would be nice if it would also toggle through the 3 sizes.

**Comment by rxhanson** [comment]
I don't have the bandwidth to add it at the moment, but I see no reason why I would reject a pull request for this.

**Reply by mastef** [reply]
Thank you for your quick reply!

I haven't coded in Swift yet, but I suppose it's using some code from here [LeftRightHalfCalculation.swift](https://github.com/rxhanson/Rectangle/blob/90d6858dbfd26b581b9af6e12f10572f25618072/Rectangle/WindowCalculation/LeftRightHalfCalculation.swift) and applying it to [CenterHalfCalculation.swift](https://github.com/rxhanson/Rectangle/blob/439e3379dffc182d0f30990f83880460c634971c/Rectangle/WindowCalculation/CenterHalfCalculation.swift).

For example making `CenterHalfCalculation` inherit from `RepeatedExecutionsInThirdsCalculation` as well. But then I only see `isRepeatedCommand` being used in `calculate*AcrossDisplays` methods - which aren't of use to us for this case I think - since we want to resize on the current screen.

So I'm thinking overriding the `calculate` method and then implementing the `.resize` case would be the right idea? https://github.com/rxhanson/Rectangle/blob/90d6858dbfd26b581b9af6e12f10572f25618072/Rectangle/WindowCalculation/LeftRightHalfCalculation.swift#L26-L29

Basically copying the `calculate` method without the first `.acrossMonitor, .acrossAndResize` case.

**Reply by rxhanson** [reply]
Thanks for looking to contribute! 

Instead of referencing `LeftRightHalfCalculation`,  look at a simpler one like `UpperLeftCalculation`, since we're not concerned with moving to another display.
